test/use-semicolon-delimiter.test.js

```diff
@@ -1,110 +1,101 @@
 'use strict'
 
-const t = require('tap')
-const test = t.test
+const { test } = require('node:test')
 const Fastify = require('..')
 const sget = require('simple-get').concat
 
 test('use semicolon delimiter default false', t => {
-  t.plan(4)
-
   const fastify = Fastify({})
 
-  t.teardown(fastify.close.bind(fastify))
+  t.after(fastify.close())
 
   fastify.get('/1234;foo=bar', (req, reply) => {
     reply.send(req.query)
   })
 
   fastify.listen({ port: 0 }, err => {
-    t.error(err)
+    t.assert.ifError(err)
 
     sget({
       method: 'GET',
       url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
     }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(body), {})
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepStrictEqual(JSON.parse(body), {})
     })
   })
 })
 
 test('use semicolon delimiter set to true', t => {
-  t.plan(4)
-
   const fastify = Fastify({
     useSemicolonDelimiter: true
   })
-  t.teardown(fastify.close.bind(fastify))
+  t.after(fastify.close())
 
   fastify.get('/1234', (req, reply) => {
     reply.send(req.query)
   })
 
   fastify.listen({ port: 0 }, err => {
-    t.error(err)
+    t.assert.ifError(err)
 
     sget({
       method: 'GET',
       url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
     }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(body), {
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepStrictEqual(JSON.parse(body), {
         foo: 'bar'
       })
     })
   })
 })
 
 test('use semicolon delimiter set to false', t => {
```

Comment:
Call `done` in after last test assertion
```suggestion
test('use semicolon delimiter set to false', (t, done) => {
  t.plan(4)
```